### PR TITLE
feat: Adds product analytics on most of the routes

### PR DESCRIPTION
**Pull request by frgfm**
This PR introduces the following modifications:
- simplifies the storage interactions (bucket/s3 --> storage.py)
- adds a telemetry client
- adds event capture on most routes

Any feedback is welcome!

cf. #304

### CODE REVIEW

**Comment by blenzi** [comment]
Thanks for this nice addition @frgfm ! I believe the errors in the CI come from 
`from app.services import telemetry_client` (not defined) vs from app.services.telemetry import telemetry_client`.

I would suggest monitoring also the alert notifications, in order to remove that from the db, as you suggested in #304.

File: src/app/api/endpoints/alerts.py [hunk 1]
```diff
@@ -81,23 +70,39 @@ async def create_alert(
...
+    background_tasks: BackgroundTasks,
+    user: Annotated[UserRead, Security(get_current_user, scopes=[AccessType.admin])],
+) -> AlertOut:
+    """
+    Creates a new alert based on the given information and send a notification if it is the first alert of the event
+
+    Below, click on "Schema" for more detailed information about arguments
+    or "Example Value" to get a concrete idea of arguments
+    """
+    telemetry_client.capture(user.id, event="alerts-create")
```
Comment (blenzi) [inline src/app/api/endpoints/alerts.py:85]: Add device / site / group name ?

File: src/app/api/endpoints/alerts.py [hunk 2]
```diff
@@ -81,23 +70,39 @@ async def create_alert(
...
     device: DeviceOut = Security(get_current_device, scopes=[AccessType.device]),
-):
+) -> AlertOut:
     """
     Creates an alert related to the authentified device, uses its device_id as argument
 
     Below, click on "Schema" for more detailed information about arguments
     or "Example Value" to get a concrete idea of arguments
     """
+    telemetry_client.capture(device.id, event="alerts-create-from-device", properties={"owner_id": device.owner_id})
```
Comment (blenzi) [inline src/app/api/endpoints/alerts.py:105]: Add device, site and group name instead ?

File: src/app/api/endpoints/devices.py [hunk 3]
```diff
@@ -26,17 +26,19 @@
...
 
 @router.post("/", response_model=DeviceOut, status_code=status.HTTP_201_CREATED, summary="Create a new device")
-async def register_device(payload: AdminDeviceAuth, _=Security(get_current_access, scopes=[AccessType.admin])):
+async def register_device(payload: AdminDeviceAuth, access=Security(get_current_access, scopes=[AccessType.admin])):
     """Creates a new device based on the given information
 
     Below, click on "Schema" for more detailed information about arguments
     or "Example Value" to get a concrete idea of arguments
     """
+    telemetry_client.capture(access.id, event="devices-create")
```
Comment (blenzi) [inline src/app/api/endpoints/devices.py:41]: Add device / site / group name ?

**Review by blenzi — COMMENTED** [review]
Thanks a lot for these additions, @frgfm ! 

I have no experience with PostHog but if I understand well, it will not have access to our db. Therefore, to maximize the possibilities it offers, we probably need some more detailed, human-readable information about the events. This will be required when analyzing individual alerts, statistics per site / group, etc. So I added in a few places the suggestion to include device / site / group name, updated information, etc. What do you think ?

**Comment by frgfm** [comment]
So apart from "updated info" (feels like dumping the payload otherwise), I added everything that was immediately available (not requiring another DB operations)

**Review by blenzi — APPROVED** [review]
Thanks @frgfm ! Let's start using it asap and adjust the level of information if needed

**Pull request merged**